packages/vscode-extension/src/builders/BuildCache.ts

```diff
@@ -30,44 +30,34 @@ export type BuildCacheInfo = {
   buildResult: AndroidBuildResult | IOSBuildResult;
 };
 
-export class PlatformBuildCache {
-  static instances: Record<DevicePlatform, PlatformBuildCache | undefined> = {
-    [DevicePlatform.Android]: undefined,
-    [DevicePlatform.IOS]: undefined,
-  };
-
-  static forPlatform(platform: DevicePlatform): PlatformBuildCache {
-    if (!this.instances[platform]) {
-      this.instances[platform] = new PlatformBuildCache(platform);
-    }
-
-    return this.instances[platform];
-  }
-
-  private constructor(private readonly platform: DevicePlatform) {}
+export class BuildCache {
+  constructor(private readonly platform: DevicePlatform, private readonly appRoot: string) {}
 
   get cacheKey() {
-    return this.platform === DevicePlatform.Android ? ANDROID_BUILD_CACHE_KEY : IOS_BUILD_CACHE_KEY;
+    const platformKey =
+      this.platform === DevicePlatform.Android ? ANDROID_BUILD_CACHE_KEY : IOS_BUILD_CACHE_KEY;
+    const appRoot = this.appRoot;
+    return appRoot + platformKey;
```

Comment:
would prefer to add some delimiter here, would be cleaner to read in case we actually need to access this in the future.

Also, because of that, it'd be better to have the path at the end, otherwise it would be harder to get the path as you couldn't tell where the file path actually ends.

Finally, lets make this `private` as it is not being accessed from the outside, and I don't think it needs to be exposed.

So I'd go for:
```
private get cacheKey() {
  const keyPrefix = this.platform === DevicePlatform.Android ? ANDROID_BUILD_CACHE_KEY : IOS_BUILD_CACHE_KEY;
  return `${keyPrefix}:{this.appRoot}`
}
```